### PR TITLE
framework/task_manager: Add pause and resume functionality and its utc

### PR DESCRIPTION
**Pull request by jeongchanKim**
1. Task manager support the pause and resume
2. Task manager utc can cover pause and resume APIs also.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [bdb69f37cb63320b12fc2c63d4a442cc325cffa1] -

**Comment by seinfra** [comment]
Target : [bdb69f37cb63320b12fc2c63d4a442cc325cffa1] -

**Comment by seinfra** [comment]
Target : [fefafc3c425d2c2fe8a5cbda78b09fad03109faa] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [fefafc3c425d2c2fe8a5cbda78b09fad03109faa] - Code Rule Check Result:
apps/examples/testcase/ta_tc/task_manager/utc/utc_task_manager_main.c:60: ERROR: [BRC_M_SMT] while should follow close brace '}'

**Comment by seinfra** [comment]
Target : [93439b145455e43344e92c4160d285dcfbfb5462] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [93439b145455e43344e92c4160d285dcfbfb5462] - Code Rule Check Result:
apps/examples/testcase/ta_tc/task_manager/utc/utc_task_manager_main.c:60: ERROR: [BRC_M_SMT] while should follow close brace '}'

**Comment by seinfra** [comment]
Target : [cdfaf46dcf43451a6be77f2927123ef1dd93ae2b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [cdfaf46dcf43451a6be77f2927123ef1dd93ae2b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b0f3142349eb20a177877d4090d7d608f7850038] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b0f3142349eb20a177877d4090d7d608f7850038] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [a2990a87d5e24a5dc8f97ab0d803b55f4f22bd61] -

**Comment by seinfra** [comment]
Target : [a2990a87d5e24a5dc8f97ab0d803b55f4f22bd61] -

**Comment by seinfra** [comment]
Target : [6a6dcb952c98754a017f3a3f337db1eb931db9a6] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6a6dcb952c98754a017f3a3f337db1eb931db9a6] -

**Comment by seinfra** [comment]
Target : [6a6dcb952c98754a017f3a3f337db1eb931db9a6] - Code Rule Check (C++) OK.

File: framework/src/task_manager/task_manager.c [hunk 1]
```diff
@@ -354,9 +354,46 @@ int task_manager(int argc, char *argv[])
...
 		case TASKMGT_PAUSE:
+			tm_update_task_state(request_msg.handle);
+			ret = tm_get_task_state(request_msg.handle);
+			if (ret != TM_TASK_STATE_RUNNING) {
+				ret = -ret;
+				break;
+			}
+			ret = tm_check_permission(request_msg.handle, request_msg.caller_pid);
+			if (ret != OK) {
+				break;
```
[the remark this answers is not in the document]
Comment (jeongchanKim) [reply framework/src/task_manager/task_manager.c]: I will change it after

**Comment by seinfra** [comment]
Target : [712610b55bbd316d60c945bdb62f5a846f6b4efe] -

**Comment by seinfra** [comment]
Target : [787ba67eb045afc11b74ba5e802e13b5b6a2cfda] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [787ba67eb045afc11b74ba5e802e13b5b6a2cfda] - Code Rule Check Result:
apps/examples/testcase/ta_tc/task_manager/utc/utc_task_manager_main.c:60: ERROR: [BRC_M_SMT] while should follow close brace '}'

**Comment by seinfra** [comment]
Target : [787ba67eb045afc11b74ba5e802e13b5b6a2cfda] -

File: apps/examples/testcase/ta_tc/task_manager/utc/utc_task_manager_main.c [hunk 2]
```diff
@@ -57,7 +57,8 @@ int tm_sample_main(int argc, char *argv[])
 	if (ret != OK) {
 		printf("ERROR : fail to set handler\n");
 	}
-	while (1);	// This will be dead at utc_task_manager_terminate_p
+	while (1) {
```
[the remark this answers is not in the document]
Comment (jeongchanKim) [reply apps/examples/testcase/ta_tc/task_manager/utc/utc_task_manager_main.c]: Done

**Comment by seinfra** [comment]
Target : [8ad0d5b2b4a139f0a37dd44ee329848a269b0318] -

**Comment by seinfra** [comment]
Target : [8ad0d5b2b4a139f0a37dd44ee329848a269b0318] -

**Comment by seinfra** [comment]
Target : [8ad0d5b2b4a139f0a37dd44ee329848a269b0318] -

**Comment by seinfra** [comment]
Target : [1930ceb9863f7374dedf38951eec23576402f39f] -

**Comment by seinfra** [comment]
Target : [1930ceb9863f7374dedf38951eec23576402f39f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [1930ceb9863f7374dedf38951eec23576402f39f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a9d773942151efde191a898dcb8f74d94d50565c] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [a9d773942151efde191a898dcb8f74d94d50565c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [faad3d69d617e7a57a85011dba02a9da3c9c0b16] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [faad3d69d617e7a57a85011dba02a9da3c9c0b16] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [1002f80de92a3f9f1b5a9fec12463a49de884d12] -

**Comment by seinfra** [comment]
Target : [1002f80de92a3f9f1b5a9fec12463a49de884d12] -

**Comment by seinfra** [comment]
Target : [d37a32f32b3f4c38041a9245e4d92b27ae779ea9] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d37a32f32b3f4c38041a9245e4d92b27ae779ea9] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [26ddbe716479bcdcd958cec5dae0d4cd9cb2ecab] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [26ddbe716479bcdcd958cec5dae0d4cd9cb2ecab] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d707c6591b8a768093600505ddab4aeff46947cf] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d707c6591b8a768093600505ddab4aeff46947cf] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [5789dd6808d75e3015f2510caf0cc8fdb44b84e5] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [5789dd6808d75e3015f2510caf0cc8fdb44b84e5] - Code Rule Check (C++) OK.

**Pull request merged**